### ISSUE TITLE
Missing source information causing "identical line collapsing" in VsCode Debug Console

### ISSUE THREAD

**Issue by tebeco** [OmniSharp/omnisharp-vscode#5040]
Hello everyone,

TLDR:
"Ominisharp console output" is not providing "Source Information" to VsCode
https://github.com/microsoft/vscode/issues/141218

## Description (using browser)
Yesterday we noticed that when debugging inside VsCode some code that tried to write to `StdOut`, these lines could be collapsed.
This is the same things that you could see on Chromium-based browser.
Remember for later that the first example is not grouping while the second one is.
![image](https://user-images.githubusercontent.com/2266487/150692059-76913aee-6ebf-4996-8e7c-b67d7028c160.png)

This can also be tempered with:
![image](https://user-images.githubusercontent.com/2266487/150692096-99cec2ce-1a39-4714-b43e-d267db973ed8.png)

## In VsCode
Javascript:
![image](https://user-images.githubusercontent.com/2266487/150692122-eb42211d-1a6e-40ba-960d-0622e7f581cc.png)

C# / F#:
![image](https://user-images.githubusercontent.com/2266487/150692145-9f2c2794-ac71-4e6f-aba6-750563d4fe71.png)
![image](https://user-images.githubusercontent.com/2266487/150693448-ba43e3e8-5a6f-4f5b-ab91-e0121494102f.png)

## Settings ?
We opened an issue on VsCode and we were hinted at:
![image](https://user-images.githubusercontent.com/2266487/150692159-84061e91-9cc7-4b05-9763-63ee4e34b8d6.png)

issue is ... we did not knew or set it, meaning we were running with the default value all along and still have 2 different output on the same settings which lead us here:

From @weinand => https://github.com/microsoft/vscode/issues/141218

> If console output provides source information (in the following screenshot on the right hand side), then lines are only collapsed if their source is identical. @isidorn is this correct?
> The following example shows that identical lines are collapsed in Node.js for the loop but not for individual console.log statements:
> ![image](https://user-images.githubusercontent.com/2266487/150692288-a5255acf-5515-4eeb-a799-3897e33d4218.png)

> If a debug extension does not provide source information, then identical are always collapsed. This seems to be the case for C# and F#.

**Comment by tebeco** [comment]
fine with me 
IIRC @333fred suggested me to post here i might have misunderstood ;)

**Comment by nohwnd** [comment]
@gregg-miskelly Please have a look.

**Comment by tebeco** [comment]
the question is ... does it only affect `debug.console.collapseIdenticalLines` (the fact that it's missing source information) ?

**Comment by gregg-miskelly** [comment]
Proving source references is optional, so, if VS Code had a place where it was requiring them (which I don't think it does), that would be a bug in VS Code.

**Comment by tebeco** [comment]
fine for me l, just double checking

**Comment by weinand** [comment]
I think we (VS Code) should consider **not to collapse lines** if there is no source info available.
